src/Orleans.Runtime/Configuration/Options/ActivationRebalancerOptions.cs

```diff
@@ -0,0 +1,167 @@
+
+    /// <summary>
+    /// The default value of <see cref="SiloNumberWeight"/>.
+    /// </summary>
+    public const double DEFAULT_SILO_NUMBER_WEIGHT = 0.1d;
+
+    /// <summary>
+    /// The maximum allowed number of activations that can be migrated at any given cycle.
+    /// </summary>
+    public int ActivationMigrationCountLimit { get; set; } = DEFAULT_ACTIVATION_MIGRATION_COUNT_LIMIT;
+
+    /// <summary>
+    /// The default value for <see cref="ActivationMigrationCountLimit"/>.
+    /// The default is practically no limit.
+    /// </summary>
+    public const int DEFAULT_ACTIVATION_MIGRATION_COUNT_LIMIT = int.MaxValue;
+}
+
+internal sealed class ActivationRebalancerOptionsValidator(
+    IOptions<ActivationRebalancerOptions> options,
+    IOptions<DeploymentLoadPublisherOptions> publisherOptions) : IConfigurationValidator
+{
+    private readonly ActivationRebalancerOptions _options = options.Value;
+    private readonly DeploymentLoadPublisherOptions _publisherOptions = publisherOptions.Value;
+
+    public void ValidateConfiguration()
+    {
+        if (_options.SessionCyclePeriod < 2 * _publisherOptions.DeploymentLoadPublisherRefreshTime)
+        {
+            throw new OrleansConfigurationException(
+                $"{nameof(ActivationRebalancerOptions.SessionCyclePeriod)} must be at least " +
+                $"{$"2 x {nameof(DeploymentLoadPublisherOptions.DeploymentLoadPublisherRefreshTime)}"}");
+        }
+
+        if (_options.MaxStaleCycles <= 0)
+        {
+            throw new OrleansConfigurationException($"{nameof(ActivationRebalancerOptions.MaxStaleCycles)} must be greater than 0");
+        }
+
+        if (_options.EntropyQuantum <= 0d || _options.EntropyQuantum > 0.1d)
+        {
+            throw new OrleansConfigurationException($"{nameof(ActivationRebalancerOptions.EntropyQuantum)} must be in greater than 0, and less or equal 0.1");
+        }
+
+        if (_options.AllowedEntropyDeviation <= 0d || _options.AllowedEntropyDeviation > 0.1d)
+        {
+            throw new OrleansConfigurationException($"{nameof(ActivationRebalancerOptions.AllowedEntropyDeviation)} must be in greater than 0, and less or equal 0.1");
+        }
+
+        if (_options.CycleNumberWeight <= 0d || _options.CycleNumberWeight > 1d)
+        {
+            throw new OrleansConfigurationException($"{nameof(ActivationRebalancerOptions.CycleNumberWeight)} must be in greater than 0, and less or equal to 1");
+        }
+
+        if (_options.SiloNumberWeight < 0d || _options.SiloNumberWeight > 1d)
+        {
+            throw new OrleansConfigurationException($"{nameof(ActivationRebalancerOptions.SiloNumberWeight)} must be in greater than or equal to 0, and less or equal to 1");
+        }
+
+        if (_options.ActivationMigrationCountLimit < 1)
+        {
+            throw new OrleansConfigurationException($"{nameof(ActivationRebalancerOptions.ActivationMigrationCountLimit)} must be greater than 0");
+        }
+    }
+}
```

Comment:
I am being linked to `throw new OrleansConfigurationException($"{nameof(ActivationRebalancerOptions.ActivationMigrationCountLimit)} must be greater than 0");` which I think is out-of-place

Reply:
ah, yeah you are right, but i'd rather keep it as its called `EntropyQuantum` - primarily because it sounds much cooler :)
The XML docs explain it in detail